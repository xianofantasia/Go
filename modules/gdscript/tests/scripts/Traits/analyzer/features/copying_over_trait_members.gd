trait SomeTrait:
	signal some_signal()
	enum {
		enum_value
	}
	enum named_enum {
		named_enum_value
	}
	var some_var = "some var"
	const some_const = "some const"
	func some_func():
		print("some func")

	class SomeInnerClass:
		func func_in_inner_trait():
			print("func in inner trait()")

class SomeClass:
	uses SomeTrait

func test():
	var using_class = SomeClass.new()
	using_class.some_signal.connect(print.bind("some signal"))
	using_class.some_signal.emit()
	print(using_class.enum_value)
	print(using_class.named_enum.named_enum_value)
	print(using_class.some_var)
	print(using_class.some_const)
	using_class.some_func()
	SomeClass.SomeInnerClass.new().func_in_inner_trait()
	print("ok")
