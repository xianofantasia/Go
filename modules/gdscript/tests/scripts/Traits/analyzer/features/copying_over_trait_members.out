GDTEST_OK
~~ WARNING at line 2: (UNUSED_SIGNAL) The signal "some_signal" is declared but never explicitly used in the class.
~~ WARNING at line 23: (UNSAFE_METHOD_ACCESS) The method "connect()" is not present on the inferred type "Variant" (but may be present on a subtype).
~~ WARNING at line 24: (UNSAFE_METHOD_ACCESS) The method "emit()" is not present on the inferred type "Variant" (but may be present on a subtype).
some signal
0
0
some var
some const
some func
func in inner trait()
ok
